fastparse/shared/src/main/scala/fastparse/parsers/Combinators.scala

```diff
@@ -341,43 +342,50 @@ object Combinators {
         del.parseRec(cfg, index) match{
           case f1: Mutable.Failure =>
             val cut1 = f1.cut
-            if (f1.cut) failMore(f1, index, cfg.logDepth, cut = true)
-            else passIfMin(cut, f1, index, ev.result(acc), count)
+            if (cut1) failMore(f1, index, cfg.logDepth, cut = true)
+            else passInRange(cut, f1, index, ev.result(acc), count)
 
           case Mutable.Success(value0, index0, traceParsers0, cut0)  =>
             p.parseRec(cfg, index0) match{
               case f2: Mutable.Failure =>
                 val cut2 = f2.cut
                 if (cut2 | cut0) failMore(f2, index0, cfg.logDepth, cut = true)
-                else passIfMin(cut | cut0, f2, index, ev.result(acc), count)
+                else passInRange(cut | cut0, f2, index, ev.result(acc), count)
 
               case Mutable.Success(value1, index1, traceParsers1, cut1)  =>
                 ev.accumulate(value1, acc)
-                rec(index1, delimiter, lastFailure, acc, cut0 | cut1, count + 1)
+                val counted = count + 1
+                if (counted < max)
+                  rec(index1, delimiter, lastFailure, acc, cut0 | cut1, counted)
+                else
+                  passInRange(cut0 | cut1, lastFailure, index1, ev.result(acc), counted)
             }
         }
       }
 
-      def passIfMin(cut: Boolean,
+      def passInRange(cut: Boolean,
                     lastFailure: Mutable.Failure,
                     finalIndex: Int,
                     acc: R,
                     count: Int) = {
-        if (count >= min) success(cfg.success, acc, finalIndex, lastFailure.traceParsers, cut)
-        else  failMore(lastFailure, index, cfg.logDepth, cut = cut)
-
+        if ((min <= count) && (count <= max)) {
```

Comment:
I'm pretty sure the `(count <= max)` is unnecessary because the tail-recursion stops when `count == max`, so both call sites above will never see this condition become true